internal/api/handlers/scheduler_handler.go

```diff
@@ -44,3 +47,35 @@ func (h *SchedulerHandler) ListSchedulers(ctx context.Context, message *api.Empt
 	}, nil
 
 }
+
+func (h *SchedulerHandler) CreateScheduler(ctx context.Context, request *api.CreateSchedulerRequest) (*api.Scheduler, error) {
+
+	scheduler := h.fromRequestToEntity(request)
+
+	scheduler, err := h.schedulerManager.CreateScheduler(ctx, scheduler)
+	if err != nil {
+		return nil, fmt.Errorf("failed create scheduler: %w", err)
```

Comment:
Shouldn't we care about the errors we're returning here? otherwise, our external requestors will only receive an "unknown error", making it hard to understand. [gRPC has a list of status](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) code that we can return, we would probably need to use the [status](https://pkg.go.dev/google.golang.org/grpc/status) and [codes](https://pkg.go.dev/google.golang.org/grpc/codes) package.

Reply:
That nice, didn't notice we had such things to work with!